packages/orm/src/Model.ts

```diff
@@ -0,0 +1,786 @@
+   *
+   * @param  {*} id - the `id` of the object to get
+   * @throws If object with id `id` doesn't exist
+   * @return {Model|null} {@link Model} instance with id `id`
+   */
+  static withId(id: ModelId) {
+    return this.get({
+      [this.idAttribute]: id,
+    });
+  }
+
+  /**
+   * Returns a boolean indicating if an entity
+   * with the id `id` exists in the state.
+   *
+   * @param  {*}  id - a value corresponding to the id attribute of the {@link Model} class.
+   * @return {Boolean} a boolean indicating if entity with `id` exists in the state
+   *
+   * @since 0.11.0
+   */
+  static idExists(id?: any) {
+    return this.exists({
+      [this.idAttribute]: id,
+    });
+  }
+
+  /**
+   * Returns a boolean indicating if an entity
+   * with the given props exists in the state.
+   *
+   * @param  {*}  props - a key-value that {@link Model} instances should have to be considered as existing.
+   * @return {Boolean} a boolean indicating if entity with `props` exists in the state
+   */
+  static exists(lookupObj: object) {
+    if (typeof this.session === "undefined") {
+      throw new Error(
+        [
+          `Tried to check if a ${this.modelName} model instance exists without a session. `,
+          "Create a session using `session = orm.session()` and call ",
+          `\`session["${this.modelName}"].exists\` instead.`,
+        ].join("")
+      );
+    }
+
+    return Boolean(this._findDatabaseRows(lookupObj).length);
+  }
+
+  /**
+   * Gets the {@link Model} instance that matches properties in `lookupObj`.
+   * Throws an error if {@link Model} if multiple records match
+   * the properties.
+   *
+   * @param  {Object} lookupObj - the properties used to match a single entity.
+   * @throws {Error} If more than one entity matches the properties in `lookupObj`.
+   * @return {Model} a {@link Model} instance that matches the properties in `lookupObj`.
+   */
+  static get(lookupObj: object) {
+    const ThisModel = this;
+
+    const rows = this._findDatabaseRows(lookupObj);
+    if (rows.length === 0) {
+      return null;
+    }
+    if (rows.length > 1) {
+      throw new Error(
+        `Expected to find a single row in \`${this.modelName}.get\`. Found ${rows.length}.`
+      );
+    }
+
+    return new ThisModel(rows[0]);
+  }
+
+  /**
+   * Gets the {@link Model} class or subclass constructor (the class that
+   * instantiated this instance).
+   *
+   * @return {Model} The {@link Model} class or subclass constructor used to instantiate
+   *                 this instance.
+   */
+  getClass(): typeof Model {
+    return this.constructor as typeof Model;
+  }
+
+  /**
+   * Gets the id value of the current instance by looking up the id attribute.
+   * @return {*} The id value of the current instance.
+   */
+  getId() {
+    return this._fields[this.getClass().idAttribute];
+  }
+
+  /**
+   * Returns a reference to the plain JS object in the store.
+   * Make sure to not mutate this.
```

Comment:
We could enforce this comment by adding a `Readonly<...>` return type or using `Object.freeze` etc 🤔